boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts

```diff
@@ -25,6 +25,9 @@
 		zephyr,dtcm = &dtcm;
 		zephyr,console = &lpuart1;
 		zephyr,shell-uart = &lpuart1;
+		zephyr,flash-controller = &s26ks512s0;
```

Comment:
for other boards flash-controller = &flexspi, but that made the fs shell sample not work. I am not sure if this is even needed in the dts

Reply:
Try changing `DT_DRV_COMPAT` in the driver to `nxp_imx_flexspi`. That should hopefully fix the issue you were seeing with the shell sample. This value should be `flexspi`

Reply:
The value should be flexspi when compared to the boards using the nor flash, but do they actually work with the shell sample?
They have nxp_imx_flexspi_nor as DT_DRV_COMPAT.

Reply:
Flash shell sample should be working on the RT platforms, @hakehuang can you please comment.

Reply:
with this PR, samples/driver/flash_shell is enabled, I run a basic check
```
*** Booting Zephyr OS build v2.7.0-rc3-304-gd198da7bb4ba  ***
Found flash controller FLEXSPI.
Flash I/O commands can be run.

```

Reply:
What happens if you do:
```
uart:~$ flash page_count
```